lib/cn/cn_tree.c

```diff
@@ -1588,13 +1643,16 @@ cn_comp_commit(struct cn_compaction_work *w)
     }
 
     kvsets = calloc(1, alloc_len);
-    if (ev(!kvsets)) {
-        w->cw_err = merr(ENOMEM);
+    cookiev = calloc(w->cw_outc, sizeof(*cookiev));
```

Comment:
nit: An opportunity here to combine the allocations of kvsets and cookiev into one.

Reply:
@alexttx maybe we could get your sweet allocator interface now that the alloc_aligned() stuff is good to go.

Reply:
I have it queued up.  Will put it up for review when the dust settles.